march_hardware/src/March4.cpp

```diff
@@ -0,0 +1,151 @@
+#include <vector>
+
+#include <ros/ros.h>
+
+#include <march_hardware/Joint.h>
+#include <march_hardware/TemperatureSensor.h>
+
+#include <march_hardware/EtherCAT/EthercatIO.h>
+
+#include <march_hardware/March4.h>
+
+namespace march4cpp
+{
+MARCH4::MARCH4()
+{
+  TemperatureSensor tempSens = TemperatureSensor(1, 0);
+  IMotionCube imc = IMotionCube(2);
+  Joint temp = Joint("test_joint", tempSens, imc);
+  jointList.push_back(temp);
+
+  ethercatMaster = new EthercatMaster(jointList, "enp2s0", this->getMaxSlaveIndex());
```

Comment:
The max slave index?
In a future rewrite we are planning on not passing a jointList to the EthercatMaster to avoid multiple copies of the same list.
@martijnvandermarel

Reply:
You can move this `if (ec_slavecount < this->maxSlaveIndex)` check to March4cpp. After this maxslaveindex is not needed in ethercatmaster :)